distributed/diagnostics/cluster_dump.py

```diff
@@ -0,0 +1,38 @@
+from datetime import datetime
+from typing import Any, Collection, Dict, Literal
+
+from distributed.diagnostics.plugin import SchedulerPlugin
+from distributed.scheduler import Scheduler
+
+
+class ClusterDump(SchedulerPlugin):
+    """Dumps cluster state prior to Scheduler shutdown
+
+    The Scheduler may shutdown in cases where it is in an error state,
+    or when it has been unexpectedly idle for long periods of time.
+    This plugin dumps the cluster state prior to Scheduler shutdown
+    for debugging purposes.
+    """
+
+    def __init__(
+        self,
+        scheduler: Scheduler,
+        url: str = None,
+        exclude: "Collection[str]" = (),
+        format_: Literal["msgpack", "yaml"] = "msgpack",
+        **storage_options: Dict[str, Any],
+    ):
+        if not url:
+            suffix = "msgpack.gz" if format_ == "msgpack" else "yaml"
```

Comment:
Just a note (since I think you can get rid of this logic anyway): `scheduler.dump_cluster_state_to_url` adds the suffix automatically: https://github.com/dask/distributed/blob/1b4d13fe4f6898bb825bd9e58dc3e795a7230be5/distributed/cluster_dump.py#L29-L39

Reply:
This should be a problem going forward as `url'`s are now typed as `str`